tests/integration/pilot/gateway_test.go

```diff
@@ -197,15 +197,16 @@ spec:
 			host:  "bar",
 		},
 	}
-	if t.Settings().EnableDualStack {
+	// additional tests for dual-stack scenario
+	if len(t.Settings().IPFamilies) > 1 {
 		additionalTestCases := []struct {
 			check echo.Checker
 			from  echo.Instances
 			host  string
 		}{
-			// apps.D hosts a dual-stack service,
+			// apps.D hosts an ipv4 service,
 			// apps.E hosts an ipv6 only service and
-			// apps.B hosts an ipv4 only service
+			// apps.B hosts a dual-stack service service
```

Comment:
Looks like B is a singleStack service - https://github.com/istio/istio/blob/c7481ac76e81e7263e8135920da564c1009583aa/pkg/test/framework/components/echo/common/deployment/echos.go#L202

Reply:
> Looks like B is a singleStack service - https://github.com/istio/istio/blob/c7481ac76e81e7263e8135920da564c1009583aa/pkg/test/framework/components/echo/common/deployment/echos.go#L202
> 

This is exactly what is being changed in this PR. With this PR, all default services will follow the kind IP family, (a to c) and the additional services (d and e) are single stack.